src/EditorFeatures/CSharp/DecompiledSource/AssemblyResolver.cs

```diff
@@ -131,8 +131,19 @@ PEFile MakePEFile(IAssemblySymbol assembly)
                 // reference assemblies should be fine here, we only need the metadata of references.
                 var reference = _parentCompilation.GetMetadataReference(assembly);
                 Log(CSharpEditorResources.Load_from_0, reference.Display);
-                return TryResolve(reference, PEStreamOptions.PrefetchMetadata)
-                    ?? new PEFile(reference.Display, PEStreamOptions.PrefetchMetadata);
+
+                var result = TryResolve(reference, PEStreamOptions.PrefetchMetadata);
+                if (result is not null)
+                {
+                    return result;
+                }
+
+                if (File.Exists(reference.Display))
```

Comment:
If you need a file path you can cast this the reference to PortableExecutableReference and then grab the file path property here; there's no guarantee that the display string here is really usable like this.